app/Repositories/PhotoRepository.php

```diff
@@ -39,18 +40,22 @@ public function __construct(AWS $aws, Registrar $registrar)
      * @param  int $signupId
      * @return \Rogue\Models\Photo|null
      */
-    public function create(array $data, $signupId)
+    public function create(array $data, Signup $signup)
```

Comment:
I think your original thought sounds reasonable, actually, but i believe this will break the `/posts` endpoint that is still sending `$signupId`